### PR TITLE
change modal to alert

### PR DESCRIPTION
**Pull request by yurii-kokhalevych**
##
screenshot from 2017-07-18 01-16-02
screenshot from 2017-07-18 01-16-23

 JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-163)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-XXX)

## Code reviewers

- [ ] @yarvanya
- [x] @denisbodnar
- [ ] @AlbaMontis

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo
## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/src/components/Message/Message.js [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
-        <Modal.Body id='contained-modal-body-lg'>
-          <p className='modalText'>{this.props.message}</p>
-        </Modal.Body>
-      </Modal>
-    );
-  } 
+    if (this.props.show) {
+      setTimeout(this.hide, 4000);
+      return (
+        <Alert bsStyle={this.props.view} onDismiss={this.hide} className='alertContainer' >
```
Comment (AlbaMontis) [inline app/src/components/Message/Message.js]: {this.props.show && ...}

File: app/src/containers/Contacts/contactActions.js [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
   request()
     .post(`${API.URL}/api/support`)
     .send(data)
     .end((err, res) => {
-      (res.status == 200) && dispatch(messageModal(JSON.parse(res.text).message)) &&
-      dispatch(showModal(true)) || dispatch(messageModal(JSON.parse(res.text).message)) &&
-      dispatch(showModal(true));
+      (res.status == 200) && dispatch(messageAlert(JSON.parse(res.text).message)) &&
+      dispatch(messageView(JSON.parse(res.text).view)) && dispatch(showAlert(true)) ||
+      dispatch(messageAlert(JSON.parse(res.text).message)) &&
```
Comment (AlbaMontis) [inline app/src/containers/Contacts/contactActions.js]: I can't find difference before || and after.

File: app/src/containers/Signup/signupActions.js [hunk 3]
```diff
@@ -25,8 +32,9 @@ export const signupUser = (email, password) => {
       .post(`${API.URL}/api/users`)
       .send(data)
       .end((err, res) => {
-        (res.status == 201) && (dispatch(messageModal(JSON.parse(res.text).message)) &&
-        dispatch(showModal(true))) && browserHistory.push('/') ||
-        dispatch(messageModal(JSON.parse(res.text).message)) && dispatch(showModal(true));
+        (res.status == 201) && (dispatch(messageAlert(JSON.parse(res.text).message, 5000)) &&
+        dispatch(messageView(JSON.parse(res.text).view)) && dispatch(showAlert(true))) &&
+        browserHistory.push('/') || dispatch(messageAlert(JSON.parse(res.text).message, 5000)) &&
+        dispatch(messageView(JSON.parse(res.text).view)) && dispatch(showAlert(true));
```
Comment (AlbaMontis) [inline app/src/containers/Signup/signupActions.js]: Can you do it without copy-paste?

File: app/src/components/Message/Message.scss [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
+.alertText {
   color: $black;
-  font-size: 35px;
+  font-size: 12px;
   text-align: center;
-  line-height: 50px;
+  line-height: 12px;
+}
+
+.alertContainer{
```
Comment (AlbaMontis) [inline app/src/components/Message/Message.scss]: Missed space.

File: app/src/components/Message/Message.scss [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
+  width:270px;
```
Comment (AlbaMontis) [inline app/src/components/Message/Message.scss]: Missed space.

File: app/src/containers/Contacts/contactActions.js [hunk 6]
```diff
@@ -1,24 +1,30 @@
...
 export const contactInfo = data => dispatch =>
+
   request()
     .post(`${API.URL}/api/support`)
     .send(data)
     .end((err, res) => {
-      (res.status == 200) && dispatch(messageModal(JSON.parse(res.text).message)) &&
-      dispatch(showModal(true)) || dispatch(messageModal(JSON.parse(res.text).message)) &&
-      dispatch(showModal(true));
+      (res.status == 200) && dispatch(messageAlert(JSON.parse(res.text).message)) &&
```
Comment (AlbaMontis) [inline app/src/containers/Contacts/contactActions.js]: Don't you handle error message here?

File: app/src/containers/Home/Home.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.login {
+  top: 100px;
+}
```
Comment (AlbaMontis) [inline app/src/containers/Home/Home.scss]: Add line. And is it ok on mobile?

File: app/src/containers/Signup/signupActions.js [hunk 8]
```diff
@@ -25,8 +32,9 @@ export const signupUser = (email, password) => {
       .post(`${API.URL}/api/users`)
       .send(data)
       .end((err, res) => {
-        (res.status == 201) && (dispatch(messageModal(JSON.parse(res.text).message)) &&
-        dispatch(showModal(true))) && browserHistory.push('/') ||
-        dispatch(messageModal(JSON.parse(res.text).message)) && dispatch(showModal(true));
+        dispatch(messageAlert(JSON.parse(res.text).message)) &&
+        dispatch(messageView(JSON.parse(res.text).view)) &&
+        dispatch(showAlert(true)),
```
Comment (AlbaMontis) [inline app/src/containers/Signup/signupActions.js]: Can you use just ; ?

File: app/src/containers/Contacts/contactReducer.js [hunk 9]
```diff
@@ -1,8 +1,10 @@
 const changeReducer = {
-  SHOW_MODAL: (action, state) =>
+  SHOW_ALERT: (state, action) =>
     Object.assign({}, state, {show: action.show}),
-  MESSAGE_MODAL: (action, state) =>
-    Object.assign({}, state, {message: action.message})
+  MESSAGE_ALERT: (state, action) =>
+    Object.assign({}, state, {message: action.message, time: action.time}),
```
Comment (AlbaMontis) [inline app/src/containers/Contacts/contactReducer.js]: Maybe use ({...state, message:....})

**Review by Vovk-Nazar — APPROVED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: app/src/components/Alerts/AlertsReduser.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+const Alert = {
```
Comment (denisbodnar) [inline app/src/components/Alerts/AlertsReduser.js]: file name AlertsReducer

File: app/src/components/EventsModalForm/editEventActions.js [hunk 11]
```diff
@@ -2,10 +2,28 @@ import { API, request } from 'src/helper';
 
 const EDIT_EVENT_SUCCESS = 'EDIT_EVENT_SUCCESS';
 const EDIT_EVENT_FAILURE = 'EDIT_EVENT_FAILURE';
+const SHOW_ALERT = 'SHOW_ALERT';
```
Comment (denisbodnar) [inline app/src/components/EventsModalForm/editEventActions.js]: Just import these actions from AlertActions, don't copy/paste

File: app/src/components/DeleteEventModal/deleteEventActions.js [hunk 12]
```diff
@@ -3,26 +3,47 @@ import { browserHistory } from 'react-router';
...
     .delete(`${API.URL}/api/event/${eventId}`)
     .send(eventId)
     .end((err, res) => {
+      dispatch(messageAlert(JSON.parse(res.text).message));
+      dispatch(messageView(JSON.parse(res.text).view));
+      dispatch(showAlert(true));
       (err || !res.ok) &&
-        dispatch(deleteEventFailure(res)) ||
-        dispatch(deleteEventSuccess(res));
+      dispatch(deleteEventFailure(JSON.parse(res.text).eror)) ||
```
Comment (denisbodnar) [inline app/src/components/DeleteEventModal/deleteEventActions.js]: eror is not good name

**Review by yarvanya — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: app/src/helper/messages.js [hunk 13]
```diff
@@ -8,5 +8,8 @@ module.exports = {
   textareaError: 'Must be at least 4 characters',
   noConnection: 'No connection with server',
   sendMessage: 'Your message is sent',
-  maxTextLength: 'Maximum 150 characters are allowed'
+  maxTextLength: 'Maximum 150 characters are allowed',
+  deleteGuest: 'Guest is delete',
```
Comment (AlbaMontis) [inline app/src/helper/messages.js]: Deleted

**Review by yarvanya — APPROVED** [review]

**Review by denisbodnar — APPROVED** [review]

**Pull request merged**